model/context.go

```diff
@@ -107,12 +108,42 @@ func truncate(s string) string {
 	return s
 }
 
+type NetworkInfo struct {
+	EffectiveType *string
+	RoundTripTime *int64
+	Downlink      *float64
+	DownlinkMax   *float64
+	SaveData      *bool
```

Comment:
Can we please separate `SaveData` out? It's not a property of the network like all the other things, it's both network-related and specific to the page-load request.

Maybe this one can be a direct field of `Page`.